pandas/core/window/ewm.py

```diff
@@ -321,17 +303,18 @@ def aggregate(self, func, *args, **kwargs):
 
     agg = aggregate
 
-    @Substitution(name="ewm", func_name="mean")
-    @Appender(_doc_template)
+    @doc(
+        doc_template,
+        window_method="ewm",
+        aggregation_description="(exponential weighted moment) mean",
+        parameters="",
+        numpy_args_kwargs=numpy_args_kwargs,
+        agg_method="mean",
+        other_see_also="",
+        notes="",
+        examples="",
+    )
```

Comment:
we now have empty sections.

on master
```
>>> help(pd.core.window.ewm.ExponentialMovingWindow.mean)
Help on function mean in module pandas.core.window.ewm:

mean(self, *args, **kwargs)
    Exponential weighted moving average.

    Parameters
    ----------
    *args, **kwargs
        Arguments and keyword arguments to be passed into func.

    Returns
    -------
    Series or DataFrame
        Return type is determined by the caller.

    See Also
    --------
    pandas.Series.ewm : Calling object with Series data.
    pandas.DataFrame.ewm : Calling object with DataFrame data.
    pandas.Series.mean : Similar method for Series.
    pandas.DataFrame.mean : Similar method for DataFrame.

>>>
```

in this PR
```
>>> help(pd.core.window.ewm.ExponentialMovingWindow.mean)
Help on function mean in module pandas.core.window.ewm:

mean(self, *args, **kwargs)
    Calculate the ewm (exponential weighted moment) mean.

    Parameters
    ----------

    *args, **kwargs
        For Numpy compatibility and has no effect on the computed value.

    Returns
    -------
    Series or DataFrame
        Return type is the same as the original object.

    See Also
    --------
    pandas.Series.ewm : Calling ewm with Series data.
    pandas.DataFrame.ewm : Calling ewm with DataFrames.
    pandas.Series.mean : Aggregating mean for Series.
    pandas.DataFrame.mean : Aggregating mean for DataFrame.

    Notes
    -----

    Examples
    --------

>>>
```

Reply:
Right, but Sphinx _does_ correctly omit the empty sections

<img width="1434" alt="Screen Shot 2021-01-19 at 11 27 15 AM" src="https://user-images.githubusercontent.com/10647082/105083131-5b8cd200-5a49-11eb-8440-64e9d671aaa0.png">

Would the empty sections be a dealbreaker? Otherwise, templating these doc strings does get a little trickier

Reply:
not at all. since the docstrings are dynamically generated, AFAIK ides generally cannot display the docstrings anyway.

Reply:
I think it would be good to try to avoid those empty sections, though. Personally, I often look at docstrings in plain text (in ipython console, notebook, ..), so IMO the plain text look is also important.

I think it should be possible to inject the title as well (and not have it in the base template)? Eg you can have a helper function to add the title to the actual content (if not empty) that can be called when passing in the content in the `@doc` decorator?

Reply:
Eg something like `@doc(..., notes=_doc_notes("... note content...")` where this function adds the header? (didn't try in practice, to be clear, so don't know how practical it is)

Reply:
Okay sounds like I need to change my templating strategy overall then to omit these empty sections (which is fine)